### PR TITLE
added function call to working queue in the full-slicer

### PR DESCRIPTION
**Pull request by lucasccordeiro**
After building the CFG data structure, function calls must be added to the working queue in the full-slicer for not slicing incorrectly programs that contain only function prototypes.

### CODE REVIEW

File: src/goto-instrument/goto_instrument_parse_options.cpp [hunk 1]
```diff
@@ -983,7 +983,9 @@ void goto_instrument_parse_optionst::instrument_goto_program()
        cmdline.isset("custom-bitvector-analysis"))
       config.ansi_c.defines.push_back("__CPROVER_CUSTOM_BITVECTOR_ANALYSIS");
 
-    status() << "Adding CPROVER library" << eom;
+    // add the library
+    status() << "Adding CPROVER library ("
+              << config.ansi_c.arch << ")" << eom;
```
Comment (tautschnig) [inline src/goto-instrument/goto_instrument_parse_options.cpp]: This is a very useful change, but I'd appreciate if it could be placed in a separate PR - it's entirely unrelated to the topic of this PR.

File: src/goto-instrument/full_slicer.cpp [hunk 2]
```diff
@@ -402,6 +402,8 @@ void full_slicert::operator()(
       const exprt &s=to_code_dead(e_it->first->code).symbol();
       decl_dead[to_symbol_expr(s).get_identifier()].push(e_it->second);
     }
+    else if(e_it->first->is_function_call())
+      add_to_queue(queue, e_it->second, e_it->first);
```
Comment (tautschnig) [inline src/goto-instrument/full_slicer.cpp]: This implies that all function calls are preserved, even when not on any path to the actual slicing criterion. I do agree that a variant of this is required for soundness (a major improvement would require changes as found, e.g., in Jens Krinke's PhD thesis), but ideally that would be limited to only those function calls on the call graph where the criterion can possibly be reached.

File: src/goto-instrument/goto_instrument_parse_options.cpp (hunk 1 above)
Comment (lucasccordeiro) [inline src/goto-instrument/goto_instrument_parse_options.cpp]: Implemented as suggested. I have just created another PR to handle this change: https://github.com/diffblue/cbmc/pull/703

**Comment by lucasccordeiro** [comment]
This is a duplicate of https://github.com/diffblue/cbmc/pull/898, which addresses a similar issue.
